np.sh

```diff
@@ -1,8 +1,18 @@
 #!/usr/bin/env sh
+
+if test 0 -ne `git status --porcelain | wc -l`; then
+	echo "Unclean working tree. Commit or stash changes first." >&2;
+	exit 128;
+fi
+
+if test "00" -ne `git fetch; git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }'`; then
```

Comment:
I think this can be a bit confusing, as it's essentially testing 0 != 0 as in testing numbers, not strings.

It could also benefit from considering the result of the git fetch, if it failed or not.